### ISSUE TITLE
Support ES7 Function Bind Syntax 

### ISSUE THREAD

**Issue by voronianski** [babel/babel-eslint#127]
```javascript
let foo = {};
function bar () {
  console.log(this);
}
foo::bar(); // Highlighted by eslint - "Error: Unexpected token"
```

**Comment by zloirock** [comment]
[Emm?](https://goo.gl/2luc5a)

**Comment by voronianski** [comment]
@zloirock I'm talking about linting

**Comment by zloirock** [comment]
Sorry, confused repos in the feed.

**Comment by zloirock** [comment]
For the interest install `babel-eslint` - this code (except `console`) passes tests without any problems.

**Comment by hzoo** [comment]
Yeah I'm not sure what the issue is since it should work fine with babel-eslint?

**Comment by voronianski** [comment]
@zloirock @hzoo my `.eslintrc`:

```
{
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "node": true
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "strict": 0,
    "quotes": [2, "single"],
    "no-underscore-dangle": 0,
    "no-unused-vars": 1,
    "no-unused-expressions": 0,
    "react/jsx-no-undef": 2,
    "new-cap": 0,
    "yoda": 0,
    "no-empty": 0,
    "no-extra-boolean-cast": 0,
    "consistent-return": 0
  }
}
```

**Comment by voronianski** [comment]
@zloirock @hzoo sorry, need to update to latest versions.

**Comment by hzoo** [comment]
As long as the version of babel is the one after this was introduced babel-eslint shouldn't have a problem - are all your packages the latest?

**Comment by hzoo** [comment]
Ok great
